### PR TITLE
MNT: set codcov to ignore docs and galleries

### PR DESCRIPTION
**Pull request by story645**
No idea if this is the right way to do this and not sure how to check, but it's very confusing to have documentation prs fail on code cov.

### CODE REVIEW

**Comment by ksunden** [comment]
I also think adding a `threshold: 0.5%` (or 1% or whatever level we wish to add) would greatly help reduce the number of red X's we get that most just ignore anyway.

I think this would go inside of the `default` entry? though have also have not looked too deeply at their config file format...

File: .github/codecov.yml [hunk 1]
```diff
@@ -1,10 +1,15 @@
 # codecov used to be able to find this anywhere, now we have to manually
 # tell it where to look
 comment: false
+ignore:
+  - doc/
+  - galleries/
+  - "test_*.py"
```
Comment (tacaswell) [inline .github/codecov.yml:7]: The coverage on the test files is probably the most important thing to measure.

**Review by tacaswell — CHANGES_REQUESTED** [review]
Please do not ignore coverage changes to the test files.

File: .github/codecov.yml (hunk 1 above)
Comment (tacaswell) [reply .github/codecov.yml]: you can not run our complete test suite on a single machine due to the platform specific tests/code and optional dependencies.  These checks are critical to make sure we do not accidentally drop coverage do to changes in CI.

Put another way, tests test our code, but what tests the tests?  Verifying that your tests actually run is about the only thing we have that covers that need.

@dopplershift has also taken this position in the past.

**Comment by tacaswell** [comment]
We have also seen issues where because we are feeding codecov from a bunch of different CI services they have the problem of deciding when they have received "enough" of the reports to be worth posting their report.  

For example, if we lose a windows job it can cause some tests (and code) to not be covered.  Sometimes CI has died / failed / upload errored out so it is never coming and sometimes it is just taking "too long" so we get a false-negative.

**Pull request closed without merging**